stores/cloudflare/src/storage/index.ts

```diff
@@ -0,0 +1,440 @@
+import { type StorageThreadType, type MessageType } from '@mastra/core/memory';
+import {
+  MastraStorage,
+  type TABLE_NAMES,
+  type StorageColumn,
+  type StorageGetMessagesArg,
+  type EvalRow,
+} from '@mastra/core/storage';
+import { TABLE_THREADS } from '@mastra/core/storage';
+import { type WorkflowRunState } from '@mastra/core/workflows';
+
+export interface CloudflareConfig {
+  accountId: string;
+  namespaceId: string;
+  apiToken: string;
+}
+
+export class CloudflareStore extends MastraStorage {
+  private accountId: string;
+  private namespaceId: string;
+  private apiToken: string;
+  private baseUrl: string;
+
+  constructor(config: CloudflareConfig) {
+    super({ name: 'Cloudflare' });
+    this.accountId = config.accountId;
+    this.namespaceId = config.namespaceId;
+    this.apiToken = config.apiToken;
+    this.baseUrl = `https://api.cloudflare.com/client/v4/accounts/${this.accountId}/storage/kv/namespaces/${this.namespaceId}`;
+  }
+
+  private async getKV(key: string): Promise<string | null> {
+    const url = `${this.baseUrl}/values/${encodeURIComponent(key)}`;
+    const res = await fetch(url, {
+      method: 'GET',
+      headers: {
+        Authorization: `Bearer ${this.apiToken}`,
+      },
+    });
```

Comment:
I notice you're using fetch for these operations. The code standard for storage adapters is to use the sdk of the storage provider. An example for cloudflare would be in the stores/vectorize folder, which uses the cloudflare sdk for its vector operations. KV operations can be done with this package as well.